io/src/openni2_grabber.cpp

```diff
@@ -556,13 +556,13 @@ pcl::io::OpenNI2Grabber::convertToXYZPointCloud (const DepthImage::Ptr& depth_im
 
   float bad_point = std::numeric_limits<float>::quiet_NaN ();
 
-  const std::uint16_t* depth_map = (const std::uint16_t*) depth_image->getData ();
+  auto* depth_map = reinterpret_cast<const std::uint16_t*>(depth_image->getData ());
```

Comment:
```suggestion
  const auto* depth_map = reinterpret_cast<const std::uint16_t*>(depth_image->getData ());
```
Let's keep const here, also for `depth_map` (2x), `rgb_buffer`, and `ir_map` below